src/Data/HugeNum.purs

```diff
@@ -73,6 +74,14 @@ timesSign Plus Plus = Plus
 timesSign Minus Minus = Plus
 timesSign _ _ = Minus
 
+instance isForeignHugeNum :: IsForeign HugeNum where
+  read f = do
+    n <- readNumber f
+    pure (fromNumber n)
+
+instance asForeignHugeNum :: AsForeign HugeNum where
+  write v = write (toNumber v)
```

Comment:
You make good points - the IsForeign instance would only work for a subset of values of HugeNum. I don't see any laws/expections defining IsForeign. Is zero information loss a presumed law? Only JS-native types can be IsForeign? I only see this statement about its presumed law: "_safely_ coerce" - I wonder if this should be updated.